lib/translate/common.h

```diff
@@ -0,0 +1,36 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef LIB_TRANSLATOR_COMMON_H
+#define LIB_TRANSLATOR_COMMON_H
+
+#include <cstdio>
+
+namespace translator {
+
+enum class StatusCode { kSuccess, kFailure };
+
+void SetDebugCallback(void (*callback)(const char*));
+void DebugCallback(const char* buffer);
+
+template <typename... Args>
+void DebugLog(const char* format, Args... args) {
+  char buffer[100];
+  snprintf(buffer, 100, format, args...);
+  DebugCallback(buffer);
+}
```

Comment:
Rather than using a variadic template, `DebugCallback` itself could be a C-style variadic function which forwards its arguments to `vsnprintf`. I'd also lean towards a bit more buffer space--say 1024 bytes.